src/Tizen.BatteryMonitor/Tizen.BatteryMonitor/BatteryMonitor.cs

```diff
@@ -0,0 +1,159 @@
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <param name="rtypes">list of resource type identifiers like BLE, WiFi, CPU etc.</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        public static IList<double> GetPowerUsageByAppForAllResources(string appID, DateTime start, DateTime end, IList<ResourceType> rtypes)
+        {
+            IntPtr bmHandle = IntPtr.Zero;
+            IList<double> batteryUsage = new List<double>();
+            int ret = Interop.BatteryMonitor.GetPowerUsageByAppForAllResource(appID, ConvertDateTimeToTimestamp(start), ConvertDateTimeToTimestamp(end), bmHandle);
+            if (((BatteryMonitorError)ret != BatteryMonitorError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage for all resources" + (BatteryMonitorError)ret);
+                throw BatteryMonitorErrorFactory.ThrowBatteryMonitorException(ret);
+            }
+
+            foreach(ResourceType type in rtypes) {
+                batteryUsage.Add(GetPowerUsageForResource(bmHandle,type));
+            }
+            ret = Interop.BatteryMonitor.Destroy(bmHandle);
+            if (((BatteryMonitorError)ret != BatteryMonitorError.None))
+            {
+                Log.Error(Globals.LogTag, "Error in Destroy handle, " + (BatteryMonitorError)ret);
+                throw BatteryMonitorErrorFactory.ThrowBatteryMonitorException(ret);
+            }
+            return batteryUsage;
+        }
+
+        /// <summary>
+        /// Gets the battery consumption in mAh(milli-Ampere hour) for the specific resource for the given application in custom interval.
+        /// </summary>
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="rtype">Identifier of resource type. BLE, WiFi, CPU etc.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the specific resource for the given application in custom interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        public static double GetPowerUsageByAppForResource(string appID, ResourceType rtype, DateTime start, DateTime end)
```

Comment:
How about overloading method?
```suggestion
        public static double GetPowerUsage(string appID, ResourceType rtype, DateTime start, DateTime end)
```